### PR TITLE
Fix dockerfile building error after upgrade of base image

### PR DESCRIPTION
**Pull request by GermanDZ**
Latest release of https://hub.docker.com/r/sonarsource/sonar-scanner-cli breaks this action as in:

image

Version history of docker image:

image

The proposed solution is to keep the GH Action building from the last known good docker image, upgrades to the base docker image should be tested in a branch and then promoted to the main branch (master in this repo)

### CODE REVIEW

**Comment by whi-tw** [comment]
Perhaps it'd be a good idea to point the action at specific released image tags rather than `5.0`: that way peoples' actions can't be rug-pulled like this!

**Comment by GermanDZ** [comment]
> Perhaps it'd be a good idea to point the version in the action at specific released image tags rather than `5.0`: that way peoples' actions can't be rug-pulled like this!

Yeap, I've changed the approach after some testing.

**Comment by antonio-masotti** [comment]
The problem is that the new base image `sonar-scanner-cli` image (tag `5.0`) changed the user. It used to be root, it's now `scanner-cli`).
image


We are also having the problem that the CI pipeline is now stuck due to permission denied

image

**Comment by GermanDZ** [comment]
> The problem is that the new base image `sonar-scanner-cli` image (tag `5.0`) changed the user. It used to be root, it's now `scanner-cli`). image
> 
> We are also having the problem that the CI pipeline is now stuck due to permission denied
> 
> image

Exactly, I've just updated the solution description.

**Review by antonio-masotti — APPROVED** [review]
lgtm

**Comment by tiwarishubham635** [comment]
When will the fix be merged? Our pipelines are also failing.

**Review by tuchidatr — APPROVED** [review]

**Review by imolinaDev — APPROVED** [review]

**Review by felipeplets — APPROVED** [review]

**Comment by felipeplets** [comment]
Same here. All pipelines failing at our company.

**Comment by jorishjanssen** [comment]
Yeah same for our company, all pipelines are blocked at the moment.

**Review by jorishjanssen — APPROVED** [review]

**Review by whi-tw — APPROVED** [review]

**Review by sourceful-karlson — APPROVED** [review]

**Review by arendondiosa — APPROVED** [review]

**Review by JavierReyesO — APPROVED** [review]

**Review by devsegur — APPROVED** [review]

**Review by vhsantos26 — APPROVED** [review]

**Comment by nbolton** [comment]
@cody-simms-sonarsource Any ETA on getting this PR merged? My options right now are disable SonarCloud for our CI (as it's blocking us) or branch this repo and use @GermanDZ's PR.

**Review by pierreluctg — APPROVED** [review]

**Review by MatthewBemis — APPROVED** [review]

**Review by CaueP — APPROVED** [review]
This is the best fix at the moment to unblock broken pipelines and give more time to a proper fix on the latest docker image.

Please merge ASAP 🙏🏻

**Review by MateSousa — APPROVED** [review]
LGTM!

**Review by gustavopergola — APPROVED** [review]

**Review by davirezendegb — APPROVED** [review]

**Review by radamanthys26 — APPROVED** [review]
Ok

**Comment by leandrocostam** [comment]
@claire-villard-sonarsource @alexandre-odoux-sonarsource

**Comment by ewanjoy** [comment]
@GermanDZ Thanks for providing the fix. Can we get it in soon as our CI is blocked by this issue. Thanks!

**Review by WesKetch — APPROVED** [review]

**Review by ewanjoy — APPROVED** [review]

**Review by bstaijen — APPROVED** [review]

**Review by jeremybanka — APPROVED** [review]

**Review by arturbani — APPROVED** [review]

**Review by adrghard — APPROVED** [review]

**Review by henriquecoimbra — APPROVED** [review]

**Review by jlindblom-godaddy — APPROVED** [review]

**Review by sveinse — APPROVED** [review]

**Review by gustv000 — APPROVED** [review]

**Review by sudofer — APPROVED** [review]

**Review by rathpc — APPROVED** [review]

**Review by pjuniorlima — APPROVED** [review]

**Review by dsnjunior — APPROVED** [review]
image
Exceptional coding acumen! 🌐

**Comment by devsegur** [comment]
I opened a revert for that problem: https://github.com/SonarSource/sonar-scanner-cli-docker/pull/226

**Review by sunny-trip-ninja — APPROVED** [review]
LGTM!

**Review by samirmaina — APPROVED** [review]

**Review by weslley75 — APPROVED** [review]

**Review by eduardo-galindez-Zeta — APPROVED** [review]

**Review by davidcroda — APPROVED** [review]

**Review by jtoss — APPROVED** [review]

**Review by guillenvd — APPROVED** [review]

**Review by phuongfi91 — APPROVED** [review]

**Review by joaolcaas — APPROVED** [review]

**Review by ramosbianca — APPROVED** [review]
simbora =D

**Comment by Rasool-Abbas** [comment]
image

**Review by anton-kim-getrev — APPROVED** [review]

**Review by mishagp — APPROVED** [review]

**Comment by anton-kim-getrev** [comment]
image

**Review by RodMing — APPROVED** [review]

**Review by jupitercl — APPROVED** [review]

**Review by this-is-tobi — APPROVED** [review]

**Review by holtjohnson — APPROVED** [review]

**Review by joaobibiano — APPROVED** [review]

**Review by kirhammer — APPROVED** [review]

**Comment by sunny-trip-ninja** [comment]
I have updated github actions file under `.github/workflow/build.yml`
and changed from `uses: SonarSource/sonarcloud-github-action@master` to `uses: SonarSource/sonarcloud-github-action@v1.9.1` worked for me

**Review by mdshack — APPROVED** [review]
this looks amazing 🚀

**Review by mahdiar-naufal-shyftplan — APPROVED** [review]

**Review by snajera-livefront — APPROVED** [review]

**Review by jtownsley — APPROVED** [review]

**Review by wellington-groundswell — APPROVED** [review]

**Review by Vilius-kul — APPROVED** [review]
lgtm

**Review by antonio-lm-jr — APPROVED** [review]

**Review by ormousta — APPROVED** [review]
LGTM! 👍

**Review by viniciusgabrielfo — APPROVED** [review]

**Review by david-livefront — APPROVED** [review]

**Comment by viniciusgabrielfo** [comment]
definitely one of the best PRs I've ever seen! LGTM

**Review by arrman-saha — APPROVED** [review]

**Review by connor-livefront — APPROVED** [review]
LGTM

**Review by k1ch — APPROVED** [review]

**Comment by milewa** [comment]
> I have updated github actions file under `.github/workflow/build.yml` and changed from `uses: SonarSource/sonarcloud-github-action@master` to `uses: SonarSource/sonarcloud-github-action@v1.9.1` worked for me

I've tried that and got an error around
```
ERROR: Error during SonarScanner execution
java.lang.IllegalStateException: The plugin [python] does not support Java 11.0.20
```
Did you have to update anything else?

**Comment by tylerthecoder** [comment]
LGTM 🥇

**Review by tylerthecoder — APPROVED** [review]

**Comment by noventa-arena** [comment]
LGTM

**Review by kconst — APPROVED** [review]

**Review by noventa-arena — APPROVED** [review]

**Review by hugosanga — APPROVED** [review]
LGTM! 🙌

**Comment by sveinse** [comment]
@sunny-trip-ninja 
> I have updated github actions file under `.github/workflow/build.yml` and changed from `uses: SonarSource/sonarcloud-github-action@master` to `uses: SonarSource/sonarcloud-github-action@v1.9.1` worked for me

Tested this. It did continue the setup step, however it fails later `java.lang.IllegalStateException: The plugin [python] does not support Java 11.0.20` :(

**Review by LeandroFMLeal — APPROVED** [review]
LGTM

**Review by adamzr — APPROVED** [review]

**Review by csrudy — APPROVED** [review]
image

**Review by colo-o — APPROVED** [review]

**Review by andresplass — APPROVED** [review]

**Review by raphaelcangucu — APPROVED** [review]

**Comment by raphaelcangucu** [comment]
Same HERE!

File: Dockerfile
Comment (adrian-livefront) [inline Dockerfile:1]: Suggestion: maybe pin it to the [hash](https://hub.docker.com/layers/sonarsource/sonar-scanner-cli/5.0.0/images/sha256-2240c1507a7b9c0ce5a8ffb9bfb94a85ec321f2e6527ced16c31dd727e099cf2?context=explore) instead of the tag?

```suggestion
FROM sonarsource/sonar-scanner-cli@sha256:2240c1507a7b9c0ce5a8ffb9bfb94a85ec321f2e6527ced16c31dd727e099cf2
```

**Review by joeatbalto — APPROVED** [review]

**Review by adrian-livefront — APPROVED** [review]

**Review by a-mufasa — APPROVED** [review]
mmm feels a little too easy

**Review by elgordomac — APPROVED** [review]
LGTM

**Review by arrmansa — APPROVED** [review]

**Comment by tovbinm** [comment]
🚢 it!

**Review by cristobalilabaca — APPROVED** [review]
🐳

**Comment by arrmansa** [comment]
c137f8c67272971c001aad747dce830d81d62dce25acd0e4b8ae7290b225657c
pretty neat rick and morty reference
if only it didn't break the pipeline

**Comment by vadymgamba** [comment]
LGTM

**Comment by tuukkasp** [comment]
👍

**Comment by DanielMukh** [comment]
Same here, all pipelines are blocked. Please merge this asap 🙏

**Review by vansoares — APPROVED** [review]

File: Dockerfile
Comment (k1ch) [inline Dockerfile:1]: Switch the `USER` instead of updating the base image. 

```suggestion
FROM sonarsource/sonar-scanner-cli:5.0

USER root
```

**Review by Ghoulsas — APPROVED** [review]

**Comment by adamzr** [comment]
Both changes are correct - this step should pin to an exact version of the parent step, so that they can be updated together. Also, the next version of this step should switch to use the new Linux user.

In the meantime, I solved this by making a fork `ymeadows/sonarcloud-github-action` where I updated to use the 5.0.0 parent image. I have switched to use that in my workflows.

**Review by DerekBoucher — APPROVED** [review]

**Review by sourceful-karlson — APPROVED** [review]

**Review by aminmahrami — APPROVED** [review]

**Review by nickmcclorey-8451 — APPROVED** [review]

**Review by praxder — APPROVED** [review]

**Comment by praxder** [comment]
Same, our pipelines are blocked right now

**Review by tate-minch-bw — APPROVED** [review]

**Comment by antonio-lm-jr** [comment]
@claire-villard-sonarsource, @dbmeneses please!

**Review by bisamov7 — COMMENTED** [review]
LGTM

**Review by allyssonsantos — APPROVED** [review]

**Comment by aminmahrami** [comment]
@csaba-feher-sonarsource @istvan-bodo-sonarsource would you please have a look at this?

**Review by AdamJackman — APPROVED** [review]

**Comment by tovbinm** [comment]
What is the latest working commit in `sonarcloud-github-action` we can use until this is  resolved?
```
- name: SonarCloud Scan
        uses: sonarsource/sonarcloud-github-action@<sha>
```

**Comment by ganncamp** [comment]
https://community.sonarsource.com/t/outage-sonarscanner-cli-docker-sonarqube-sonarcloud-github-actions/113548/29

**Review by myjokersss — APPROVED** [review]
Ship it!

**Review by samoldenburg — APPROVED** [review]

File: Dockerfile
Comment (AntonyF-Andreani) [reply Dockerfile:1]: The versions apparently published are:

image


So I dont know if that's going to work (the original PR)

**Review by adriangomez24 — APPROVED** [review]

**Review by tophercf — APPROVED** [review]

**Review by gpbPiazza — APPROVED** [review]

**Review by andredallacosta — APPROVED** [review]

**Review by nicklindsley — APPROVED** [review]

**Review by ratebaltal — APPROVED** [review]

**Comment by dbmeneses** [comment]
Docker image v5.0.1 has been fixed

**Comment by Ilyes512** [comment]
How can we prevent this from happening again? Would have pinning the https://github.com/SonarSource/sonarcloud-github-action/blob/master/Dockerfile#L1 to a specific @sha have helped perhaps?

**Comment by whi-tw** [comment]
@dbmeneses That's not particularly comforting, as another bad release from sonarsource will have the same effect.

Can I suggest coupling action:image versions, so people are able to pin the action to a certain tag (and therefore docker image tag) and can reliably always run the same version of the tooling every run.

Currently, if a bad actor was able to push a malicious release onto your docker hub tag, all the action consumers would immediately start running that bad code, regardless of whether they've pinned the action.

People who specify specific versions (or commit SHAs) of an action do so to insure that no upstream changes will affect the software running in their pipelines.

Unfortunately this incident has illustrated that y'all are able to inject different software at will, without the repo owner having any real control or visibility over this. That doesn't feel particularly great, especially as we need to pass a GITHUB_TOKEN into the action: potentially giving any malicious injection access to the entire repository.

**Comment by biohazd** [comment]
Question: how do we mitigate against this breaking a pipeline? Do use a known good version of the code ?

**Comment by giorilla** [comment]
Funny how a Code Quality company just broke their tooling for all their users with a change that was obviously not tested thoroughly.

**Comment by pierre-guillot-gh** [comment]
Hello, you can find an official report summary [here](https://community.sonarsource.com/t/outage-sonarscanner-cli-docker-sonarqube-sonarcloud-github-actions/113548/36?u=pierreguillot). It contains context about the incident and the remediation actions we took to ensure it does not happen again.

**Comment by whi-tw** [comment]
Thank you for transparency - super appreciated.

**Pull request closed without merging**
